.github/workflows/javadoc-publication.yaml

```diff
@@ -0,0 +1,104 @@
+      - name: Keep only javadoc.jar
+        run: find ${{ steps.local_repo.outputs.dir }} -type f ! -name "*-javadoc.jar" -delete
+      - name: List artifacts
+        run: ls ${{ steps.local_repo.outputs.dir }}
+      - name: Create javadoc dir
+        run: mkdir -p ${{ steps.local_repo.outputs.dir }}/javadoc/${{ github.event.release.tag_name }}
+      - name: Extract javadoc.jar
+        run: cd ${{ steps.local_repo.outputs.dir }} && mv *javadoc.jar javadoc.zip && unzip javadoc.zip -d javadoc/${{ github.event.release.tag_name }}
+      - name: List javadoc files
+        run: ls "${{ steps.local_repo.outputs.dir }}/javadoc/${{ github.event.release.tag_name }}"
+      - name: Publish javadoc files to S3
+        uses: SonarSource/gh-action_release/aws-s3@feature/svermeille/BUILD-4036-Automate-javadoc-deployment
+        with:
+          command: cp
+          flags: --recursive
+          source: ${{ steps.local_repo.outputs.dir }}/javadoc
+          destination: s3://javadocs-cdn-eu-central-1-prod/${{ inputs.javadocDestinationDirectory }}
+          aws_access_key_id: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_access_key_id }}
+          aws_secret_access_key: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_secret_access_key }}
+          aws_session_token: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_security_token }}
+          aws_region: eu-central-1
+      - name: Delete dir named latest in S3
+        uses: SonarSource/gh-action_release/aws-s3@feature/svermeille/BUILD-4036-Automate-javadoc-deployment
+        with:
+          command: rm
+          destination: s3://javadocs-cdn-eu-central-1-prod/${{ inputs.javadocDestinationDirectory }}/latest
+          aws_access_key_id: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_access_key_id }}
+          aws_secret_access_key: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_secret_access_key }}
+          aws_session_token: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_security_token }}
+          aws_region: eu-central-1
+        continue-on-error: true # the first time a project publish javadoc, there is no latest dir available
+      - name: Upload to dir named latest in S3
+        uses: SonarSource/gh-action_release/aws-s3@feature/svermeille/BUILD-4036-Automate-javadoc-deployment
+        with:
+          command: cp
+          flags: --recursive
+          source: ${{ steps.local_repo.outputs.dir }}/javadoc/${{ github.event.release.tag_name }}/
+          destination: s3://javadocs-cdn-eu-central-1-prod/${{ inputs.javadocDestinationDirectory }}/latest
+          aws_access_key_id: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_access_key_id }}
+          aws_secret_access_key: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_secret_access_key }}
+          aws_session_token: ${{ fromJSON(steps.secrets.outputs.vault).javadoc_aws_security_token }}
+          aws_region: eu-central-1
```

Comment:
I would move all this to a single call to an "upload-javadoc" action instead of one step per Shell command and one call to aws-s3 per s3 command.

Reply:
I prefer to keep it this way as I see it easier to troubleshoot in case of failure and this allows to reuse aws-s3 action as a component. I see it more "evolutive" this way.

Such change is not impacting the behaviour. As you are in holidays and this is blocking the sprint, I will split it as a new ticket https://sonarsource.atlassian.net/browse/BUILD-4546.

Could you please update the ticket and share what is the advantage of hiding all this logic in a single action from your perspective ?